### PR TITLE
Task 3

### PR DESCRIPTION
**Pull request by bioneoficial**

### CODE REVIEW

File: games_report.txt [hunk 1]
```diff
@@ -0,0 +1,177 @@
+Game 2:
```
Comment (halan) [inline games_report.txt]: You got the report info, but the format is different than the expected (as json)

File: means_of_death.rb [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    20 => 'MOD_SUICIDE',
+    21 => 'MOD_TARGET_LASER',
+    22 => 'MOD_TRIGGER_HURT',
+    23 => 'MOD_NAIL',
+    24 => 'MOD_CHAINGUN',
+    25 => 'MOD_PROXIMITY_MINE',
+    26 => 'MOD_KAMIKAZE',
+    27 => 'MOD_JUICED',
+    28 => 'MOD_GRAPPLE'
+  }.freeze
```
Comment (halan) [inline means_of_death.rb:34]: excellent

File: spec/game_parser_spec.rb [hunk 3]
```diff
@@ -1,19 +1,35 @@
...
+require_relative '../top_kill_report'
 
 describe GameParser do
   let(:parser) { GameParser.new }
+  let(:file) { double(:file) }
   let(:games_file) { 'games.log' }
 
+  describe 'Creates new GameParser instance' do
+    it 'creates a new GameParser instance' do
+      expect(GameParser.new).to be_an_instance_of(GameParser)
```
Comment (halan) [inline spec/game_parser_spec.rb]: Why not use the let parser here?

File: spec/game_parser_spec.rb [hunk 4]
```diff
@@ -1,19 +1,35 @@
...
+    end
+  end
   describe '#parse' do
     it 'parses the games log and creates a hash of games' do
       expect(parser.parse(games_file)).to be_an_instance_of(Hash)
     end
 
+    it 'should remove games with no players' do
+      allow(File).to receive(:open).with(file, 'r').and_yield(file)
```
Comment (halan) [inline spec/game_parser_spec.rb]: I recommend using a fixture instead mocking it. That way your specs is requiring a much strict concern on how to open and read the file.

File: spec/game_parser_spec.rb [hunk 5]
```diff
@@ -1,19 +1,35 @@
...
+      allow(file).to receive(:each_line).and_return([])
+      allow(parser).to receive(:process_game_lines)
+      parser.parse(file)
+      expect(parser.games).to receive(:reject!)
```
Comment (halan) [inline spec/game_parser_spec.rb:27]: Take care to don't build specs requiring strict details on the method itself, and there are several solutions to reach the same result, try to write specs able not broken if you change the implementation without changing the results.

File: spec/game_spec.rb [hunk 6]
```diff
@@ -8,6 +8,11 @@
   let(:game) { Game.new }
   let(:line) { '2:22 Kill: 3 2 10: Isgalamido killed Dono da Bola by MOD_RAILGUN' }
 
+  describe '#initialize' do
+    it 'creates a new game instance' do
+      expect(Game.new).to be_an_instance_of(Game)
```
Comment (halan) [inline spec/game_spec.rb]: You already did it on let `:game`. Just use it.

Comment (halan) [reply spec/game_spec.rb]: Also, to check if `expect(Anything.new).to be_an_instance_of(Anything)` isn't testing your code. You're testing if Ruby can create instances. That spec will not help you at all.

File: spec/player_spec.rb [hunk 7]
```diff
@@ -23,4 +29,44 @@
...
+      expect { player0.decrement_kill }.to change { player0.kills }.by(-1)
+    end
+
+    it 'decrements kills by 10' do
+      10.times { player1.decrement_kill }
+      expect(player1.kills).to eq(-10)
+    end
+
+    it 'decrements kills by a random number' do
+      random_number = rand(1..20)
```
Comment (halan) [inline spec/player_spec.rb]: Be careful with random on your specs in favor of using repeatable examples.

File: games_report.txt (hunk 1 above)
Comment (bioneoficial) [reply games_report.txt]: so, to fix it i need to write this file in json synthax, yes?

Comment (halan) [reply games_report.txt]: https://gist.github.com/talyssonoc/a51b5a6c42c8c60c0f8c0f28b34f8dbc

> Para cada jogo o parser deve gerar algo como:
```
game_1: {
    total_kills: 45;
    players: ["Dono da bola", "Isgalamido", "Zeh"]
    kills: {
      "Dono da bola": 5,
      "Isgalamido": 18,
      "Zeh": 20
    }
  }
  ```
  
  You should strictly follow the requirements.

Comment (halan) [reply games_report.txt]: I mean, you did a great job, and your report looks good. But in our case, the format is part of the requirements.

**Comment by bioneoficial** [comment]
@halan  applied the suggestions, can i merge?

**Review by ghost — APPROVED** [review]

**Pull request merged**